completions/java

```diff
@@ -114,7 +114,7 @@ _comp_cmd_java__packages()
     local -a sourcepaths=("${REPLY[@]}")
 
     local REPLY
-    _comp_dequote "$cur"
+    _comp_dequote_incomplete "$cur"
```

Comment:
It's not immediately clear to me why we'd do this when completing java packages, the commit messages nor the referenced issues don't mention it either, and there's seemingly no test suite coverage for it. Is it necessary, and if yes, can we somehow clarify why and add test cases?